nbody/nbody_unsafe_simd.jl

```diff
@@ -0,0 +1,234 @@
+# Based on https://benchmarksgame-team.pages.debian.net/benchmarksgame/program/nbody-rust-7.html
+
+module NBody
+
+using StaticArrays, SIMD, Printf
+using Base: llvmcall
+
+const solar_mass = 4π^2
+const days_per_year = 365.24
+const NBODIES = 5
+const NPAIRS = Int(NBODIES * (NBODIES - 1) / 2)
+const PAIRS = Tuple((i,j) for i = 1:5, j = 1:5 if j > i)
+
+struct Bodies
+    x::MMatrix{NBODIES, 3, Float64}
```

Comment:
This is cheating imo. Are other languages really encoding the number of bodies as a compile time constant?

Reply:
Very much so, e.g.:

- https://benchmarksgame-team.pages.debian.net/benchmarksgame/program/nbody-rust-7.html
- `#define BODIES_COUNT 5`: https://benchmarksgame-team.pages.debian.net/benchmarksgame/program/nbody-gcc-8.html

I don't have a strong opinion about this, although it seems inevitable that benchmark implementations slowly slide towards more compile-time computation in the absence of clear, explicit guidelines.